src/client/gui/lib/settings/autostart_notifiers.dart

```diff
@@ -0,0 +1,83 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:flutter/services.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+final autostartProvider =
+    AsyncNotifierProvider.autoDispose<AutostartNotifier, bool>(() {
+  if (Platform.isLinux) return LinuxAutostartNotifier();
+  if (Platform.isWindows) return WindowsAutostartNotifier();
+  if (Platform.isMacOS) return MacOSAutostartNotifier();
+  throw UnimplementedError('no autostart notifier for this platform');
+});
+
+abstract class AutostartNotifier extends AutoDisposeAsyncNotifier<bool> {
+  Future<void> set(bool value) async {
+    try {
+      await _doSet(value);
+    } finally {
+      ref.invalidateSelf();
+    }
+  }
+
+  Future<void> _doSet(bool value);
+}
+
+class LinuxAutostartNotifier extends AutostartNotifier {
+  static const autostartFile = 'multipass.gui.autostart.desktop';
+  final file = File(
+    '${Platform.environment['HOME']}/.config/autostart/$autostartFile',
+  );
+
+  @override
+  Future<bool> build() => file.exists();
+
+  @override
+  Future<void> _doSet(bool value) async {
+    if (value) {
+      final data = await rootBundle.load('assets/$autostartFile');
+      await file.writeAsBytes(data.buffer.asUint8List());
+    } else {
+      if (await file.exists()) await file.delete();
+    }
+  }
+}
+
+class WindowsAutostartNotifier extends AutostartNotifier {
+  final link = Link(
```

Comment:
Can we rely on links being enabled on Windows? Or does this create a Windows shortcut?

Reply:
Hmm, I didn't know symlinks could be disabled on Windows :/ I don't think I did anything special to enable them, so I'm not sure how it would work by default.
It does not create a shortcut. I'll try to see if I can disable symlinks and test how it reacts in that case.

Reply:
That may be why I didn't go this route when implementing autostart. I believe symlinks are disabled by default. Do you remember enabling them, perhaps when you first started working with our git repo there? Anyway, we can't expect users to have them on. However, that autostart folder may work with windows shortcuts. But yeah, better check first.